qos_guidelines.adoc

```diff
@@ -133,72 +133,71 @@ accessed.
-that all of that execution is attributed to the new VM. Further in this context
-switch process, if the hypervisor intends some of the execution to be attributed
-to neither the outgoing VM nor the incoming VM, then the hypervisor may switch
-to a new configuration that is different from the configuration of either of the
-VMs for the duration of such execution. QoS capabilities are statistical in
-nature and the small duration, such as the few instructions in the hypervisor
-trap handler entrypoint, for which the HS-mode may execute with the `RCID`/
-`MCID` established for lower privilege mode operation may not be statistically
-significant.
+A Hypervisor might cause a context switch from one virtual machine to another. The
+context switch usually involves saving the context associated with the VM that is being
+switched away from and restoring the context of the VM that is being switched to. Such
+context switch might be invoked in response to an explicit call from the VM (for example,
+as a function of an `ECALL` invocation) or might be done asynchronously (for example, in
+response to a timer interrupt). In such cases, the hypervisor might want to execute
+with the `sqoscfg` configurations of the VM that is being switched away from so that
+the execution is attributed to that VM. Then, before executing the context switch code
+associated with restoring the new VMs context, first switch to the `sqoscfg` tht is 
+appropriate for the new VM that is being switched to so that all of that execution 
+is attributed to the new VM. Further in this context switch process, if the hypervisor 
+intends some of the execution to be attributed to neither the outgoing VM nor the 
+incoming VM, then the hypervisor can switch to a new configuration that is different from 
+the configuration of either of the VMs for the duration of such execution. QoS 
+capabilities are statistical in nature and the small duration, such as the few 
+instructions in the hypervisor trap handler entrypoint for which the HS-mode can 
+execute with the `RCID`/`MCID` values established for lower privilege mode operation,
+might not be statistically significant.
 
 === QoS Identifiers for supervisor and machine mode
 
-The `RCID` and `MCID` configured in `sqoscfg` also apply to execution in
-S/HS-mode but is typically not an issue. Usually, S/HS-mode execution occurs to
+The `RCID` and `MCID` values configured in `sqoscfg` also apply to execution in
+S/HS-mode, but is typically not an issue. Usually, S/HS-mode execution occurs to
 provide services, such as through the SBI, to software executing at lower
-privilege. Since the S/HS-mode invocation was to provide a service for the
-lower privilege mode, the S/HS-mode software may not modify the `sqoscfg` CSR.
+privilege. Because the S/HS-mode invocation was to provide a service for the
+lower privilege mode, the S/HS-mode software cannot modify the `sqoscfg` CSR.
```

Comment:
@kersten1 - the original sentence "may  not modify" was to indicate that as a  policy the S/HS-mode software may continue execution without modifying the `sqoscfg` CSR. It can always modify it - but it may choose to not.